src/documents/consumer.py

```diff
@@ -474,7 +475,18 @@ def progress_callback(current_progress, max_progress):  # pragma: no cover
                 ConsumerStatusShortMessage.PARSING_DOCUMENT,
             )
             self.log.debug(f"Parsing {self.filename}...")
-            document_parser.parse(self.working_copy, mime_type, self.filename)
+            if (
+                isinstance(document_parser, MailDocumentParser)
+                and self.input_doc.mailrule_id
+            ):
+                document_parser.parse(
+                    self.working_copy,
+                    mime_type,
+                    self.filename,
+                    self.input_doc.mailrule_id,
+                )
+            else:
+                document_parser.parse(self.working_copy, mime_type, self.filename)
```

Comment:
```suggestion
            document_parser.parse(
                document_path=self.working_copy,
                mime_type=mime_type,
                file_name=self.filename,
                mailrule_id=self.input_doc.mailrule_id,
            )
```

I'm trying to think if we need to pass the rule id here (or can get it later) but either way, just pass this along without the if/else, `self.input_doc.mailrule_id` defaults to None, so I *think* this will work.

Reply:
Actually, come to think of it. Changing the declaration of parser.parse is sort of problematic, it would need to be changed everywhere. You have circumvented this with `isinstance(document_parser, MailDocumentParser)`

Need to think about this one but I think we need a better solution than this (and not my suggestion)

Reply:
@stumpylog any thoughts on this question? (no rush)

Reply:
Yes this is exactly why I used `isinstance`. Without it, I broke the other passers in my first try.

Reply:
Yea, but this feels like a hack, or a change that will affect a lot of code. I'll think on it

Reply:
mostly asking stumpylog, how about this: e3119507bbe96262339868cbbc13eb2b795ae7bf dunno if that's a little hack-y but otherwise I dont see a way besides altering the parser declaration (or leaving as it currently is).

Again, no rush at all.

Reply:
Something I have been thinking about is passing parser specific settings to the parser.  So I would be fine altering the declaration of `parse` with the other ones not using it (yet)

Reply:
Great thanks stumpy. This PR is probably ready, for whenever we start merging our next round of enhancements